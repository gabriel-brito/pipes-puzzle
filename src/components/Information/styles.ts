import styled from 'styled-components'

type ButtonTypes = {
  isBlocked: boolean
  currentLevel: boolean
}

export const Wrapper = styled.section`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 32px auto;
`

export const Heading = styled.h1`
  font-size: var(--headingSize);
  margin-bottom: 32px;
`

export const Levels = styled.nav`
  background-color: var(--bgContrast);
  border-radius: 8px;
  padding: 16px;
`

export const LevelList = styled.ul`
  display: grid;
  grid-gap: 16px;
  grid-template-columns: repeat(6, 36px);
  list-style: none;
`

export const LevelListItem = styled.li`
  height: 45px;
`

export const Button = styled('button')<ButtonTypes>`
  background-color: ${(props) =>
    props.isBlocked ? 'var(--disabledBg)' : 'var(--buttonBg)'};
  background: ${(props) =>
    props.currentLevel
      ? 'linear-gradient(135deg, #ff80bf 0, #9580ff 100%)'
      : ''};
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: ${(props) => (props.isBlocked ? 'initial' : 'pointer')};
  font-size: 600;
  height: 100%;
  opacity: ${(props) => (props.currentLevel ? '0.8' : 'initial')};
  width: 100%;

  &:hover {
    opacity: ${(props) => (props.isBlocked ? 'initial' : '0.7')};
  }
`
