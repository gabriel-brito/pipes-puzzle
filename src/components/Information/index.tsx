import * as S from './styles'

type InformationTypes = {
  currentLevel: number
}

export default function Information({ currentLevel }: InformationTypes) {
  const levels = [1, 2, 3, 4, 5, 6]

  return (
    <S.Wrapper data-testid="information-wrapper">
      <S.Heading>Pipes Puzzle</S.Heading>

      <S.Levels>
        <S.Subtitle>Choose your level!</S.Subtitle>

        <S.LevelList>
          {levels.map((level) => {
            const isCurrentLevel = level === currentLevel
            const isBlocked = level > currentLevel

            return (
              <S.LevelListItem key={`level-${level}`}>
                <S.Button
                  aria-label="select-level-button"
                  currentLevel={isCurrentLevel}
                  disabled={isBlocked}
                  isBlocked={isBlocked}
                >
                  {level}
                </S.Button>
              </S.LevelListItem>
            )
          })}
        </S.LevelList>

        <S.Subtitle>Current level: {currentLevel}</S.Subtitle>
      </S.Levels>
    </S.Wrapper>
  )
}
