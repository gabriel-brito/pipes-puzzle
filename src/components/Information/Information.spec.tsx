import { screen, render } from '@testing-library/react'
import Information from '.'
import 'jest-styled-components'

describe('Information', () => {
  it('should render properly', () => {
    render(<Information currentLevel={3} />)

    expect(
      screen.getByRole('heading', { name: /pipes puzzle/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /choose your level/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /current level/i })
    ).toBeInTheDocument()
    expect(screen.queryAllByLabelText('select-level-button')).toHaveLength(6)

    expect(screen.getByTestId('information-wrapper')).toMatchSnapshot()
  })

  it('should have two disabled buttons', () => {
    render(<Information currentLevel={4} />)
    let disabled = 0

    const buttons = screen.queryAllByLabelText('select-level-button')

    buttons.forEach((button) => {
      if (button.hasAttribute('disabled')) disabled += 1
    })

    expect(disabled).toBe(2)
  })
})
