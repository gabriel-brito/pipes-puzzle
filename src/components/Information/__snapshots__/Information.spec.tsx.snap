// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information should render properly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 32px auto;
}

.c1 {
  font-size: var(--headingSize);
  margin-bottom: 32px;
}

.c3 {
  font-size: 14px;
  margin: 0 auto;
  text-align: center;
}

.c2 {
  background-color: var(--bgContrast);
  border-radius: 8px;
  padding: 16px;
}

.c4 {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: repeat(6,36px);
  list-style: none;
  margin: 16px;
}

.c5 {
  height: 45px;
}

.c6 {
  background-color: var(--buttonBg);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 600;
  height: 100%;
  opacity: initial;
  width: 100%;
}

.c6:hover {
  opacity: 0.7;
}

.c7 {
  background-color: var(--buttonBg);
  background: linear-gradient(135deg,#ff80bf 0,#9580ff 100%);
  border-radius: 8px;
  border: none;
  color: var(--buttonColor);
  cursor: pointer;
  font-size: 600;
  height: 100%;
  opacity: 0.8;
  width: 100%;
}

.c7:hover {
  opacity: 0.7;
}

<section
  class="c0"
  data-testid="information-wrapper"
>
  <h1
    class="c1"
  >
    Pipes Puzzle
  </h1>
  <nav
    class="c2"
  >
    <h2
      class="c3"
    >
      Choose your level!
    </h2>
    <ul
      class="c4"
    >
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c6"
        >
          1
        </button>
      </li>
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c6"
        >
          2
        </button>
      </li>
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c7"
        >
          3
        </button>
      </li>
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c6"
        >
          4
        </button>
      </li>
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c6"
        >
          5
        </button>
      </li>
      <li
        class="c5"
      >
        <button
          aria-label="select-level-button"
          class="c6"
        >
          6
        </button>
      </li>
    </ul>
    <h2
      class="c3"
    >
      Current level: 
      3
    </h2>
  </nav>
</section>
`;
